src/main/java/teammates/ui/controller/InstructorCourseInstructorDeleteAction.java

```diff
@@ -27,20 +27,20 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         new GateKeeper().verifyAccessible(
                 instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
 
+        // Check if there is a joined instructor (other than the instructor to delete) with the privilege of modifying instructors
         List<InstructorAttributes> instructors = logic.getInstructorsForCourse(courseId);
-        int numOfInstrCanModifyInstructor = 0;
-        InstructorAttributes instrCanModifyInstructor = null;
+        boolean hasInstrCanModifyInstructor = false;
         for (InstructorAttributes instr : instructors) {
-            if (instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR)) {
-                numOfInstrCanModifyInstructor++;
-                instrCanModifyInstructor = instr;
+            if (instr.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR)
+                && !instr.getEmail().equals(instructorEmail)
+                && instr.isRegistered()) {
```

Comment:
Introduce a local variable to explain the meaning of this long boolean condition.